bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -694,6 +722,24 @@ protected void updateChannel(String channelId) {
                     newState = new PercentType(value);
                 }
                 break;
+            case BASS:
+                value = getBass();
+                if (value != null && !isOutputLevelFixed()) {
+                    newState = new DecimalType(value);
+                }
+                break;
+            case TREBLE:
+                value = getTreble();
+                if (value != null && !isOutputLevelFixed()) {
+                    newState = new DecimalType(value);
+                }
+                break;
+            case LOUDNESS:
+                value = getLoudness();
+                if (value != null && !isOutputLevelFixed()) {
+                    newState = isLoudnessEnabled() ? OnOffType.ON : OnOffType.OFF;
```

Comment:
FYI: You could simplify to:
```suggestion
                    newState = OnOffType.from(getLoudness());
```

Reply:
Done.
Never used that until today.